tests/functional/aws-node-sdk/lib/utility/website-util.js

```diff
@@ -1,6 +1,206 @@
+    let authCredentials;
+    if (auth === 'validAuth') {
+        authCredentials = awsCredentials || {
+            accessKey: 'accessKey1',
+            secretKey: 'verySecretKey1',
+        };
+    } else if (auth === 'invalidAuth') {
+        authCredentials = {
+            accessKey: 'fakeKey1',
+            secretKey: 'fakeSecretKey1',
+        };
+    } else if (!auth) {
+        authCredentials = undefined;
+    } else {
+        throw new Error(`Unsupported auth type ${auth}`);
+    }
+    const { hostname, port, path } = url.parse(urlstring);
+    makeRequest({ hostname, port, method, path, authCredentials }, callback);
+}
+
+function _assertResponseHtml(response, elemtag, content) {
+    if (elemtag === 'ul') {
+        const startIndex = response.indexOf('<ul>');
+        const endIndex = response.indexOf('</ul>');
+        assert(startIndex > -1 && endIndex > -1, 'Did not find ul element');
+        const ulElem = response.slice(startIndex + 4, endIndex);
+        content.forEach(item => {
+            _assertResponseHtml(ulElem, 'li', item);
+        });
+    } else {
+        const elem = `<${elemtag}>${content}</${elemtag}>`;
+        assert(response.includes(elem),
+            `Expected but did not find '${elem}' in html`);
+    }
+}
+
+function _assertContainsHtml(responseBody) {
+    assert(responseBody.startsWith('<html>') &&
+        responseBody.includes('</html>'), 'Did not find html tags');
+}
+
+function _assertResponseHtml404(method, response, type, bucketName) {
+    assert.strictEqual(response.statusCode, 404);
+    if (method === 'HEAD') {
+        if (type !== '404-no-such-bucket'
+        && type !== '404-no-such-website-configuration'
+        && type !== '404-not-found') {
+            throw new Error('This 404 error is not checked in ' +
+            'checkHTML()');
+        }
+        // don't need to check HTML for head requests
+        return;
+    }
+    _assertContainsHtml(response.body);
+    _assertResponseHtml(response.body, 'title', '404 Not Found');
+    _assertResponseHtml(response.body, 'h1', '404 Not Found');
+    if (type === '404-no-such-bucket') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchBucket',
+            'Message: The specified bucket does not exist.',
+            `BucketName: ${bucketName}`,
+        ]);
+    } else if (type === '404-no-such-website-configuration') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchWebsiteConfiguration',
+            'Message: The specified bucket does not have a ' +
+            'website configuration',
+            `BucketName: ${bucketName}`,
+        ]);
+    } else if (type === '404-not-found') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchKey',
+            'Message: The specified key does not exist.',
+        ]);
+    } else {
+        throw new Error('This 404 error is not checked in ' +
```

Comment:
would it be accurate to say 'This 404 error is not a recognized 404 error in the checkHTML function.'?